libfdproto/messages.c

```diff
@@ -1646,11 +1647,44 @@ int fd_msg_avp_value_encode ( void *data, struct avp *avp )
 		
 		/* Then retrieve information about the parent's type (= derived type) */
 		CHECK_FCT(  fd_dict_getdict( avp->avp_model, &dict )  );
-		CHECK_FCT(  fd_dict_search( dict, DICT_TYPE, TYPE_OF_AVP, avp->avp_model, &parenttype, EINVAL)  );
-		CHECK_FCT(  fd_dict_getval(parenttype, &type_data)  );
+		fd_dict_search( dict, DICT_TYPE, TYPE_OF_AVP, avp->avp_model, &parenttype, EINVAL);
```

Comment:
Here please keep the CHECK_FCT( ... ) macro for other error checking, and please replace EINVAL with 0 in the parameters so not finding a parent type will not cause to stop here.